src/core/jit/exec.rs

```diff
@@ -147,19 +164,8 @@ where
                 result.map_err(|e| Positioned::new(e, field.pos)),
             );
         } else {
-            // if the present field doesn't have IR, still go through it's extensions to see
-            // if they've IR.
-            join_all(field.nested_iter().map(|child| {
-                let value = ctx.value().map(|v| v.get_key(&child.name).unwrap_or(v));
-                let ctx = if let Some(v) = value {
-                    ctx.with_value_and_field(v, child)
-                } else {
-                    ctx.with_field(child)
-                };
-                let data_path = data_path.clone();
-                async move { self.execute(child, &ctx, data_path).await }
-            }))
-            .await;
+            self.execute_inner(field, ctx, data_path.clone(), ctx.value())
```

Comment:
seems like we handling additional case when there is not IR and the value is a list. We should add an integration test